lib/connections/service.go

```diff
@@ -237,6 +224,25 @@ next:
 	}
 }
 
+// If rate limiting is set, and based on the address we
+// limit the connection, then we wrap it in a limiter.
+func (s *Service) setupReaderAndWriterWithOptionalRatelimit(c IntermediateConnection, writeRateLimit, readRateLimit *ratelimit.Bucket, options config.OptionsConfiguration) (io.Reader, io.Writer) {
```

Comment:
Explanation: I'm not sure, if this is right way. It is utility method in the scope of this service. So here the `(s *Service)` could be ommitted. I miss the `private static methodname()`-feature from java here. Hope you understand.